pkg/gits/github.go

```diff
@@ -791,25 +791,51 @@ func (p *GitHubProvider) ListOpenPullRequests(owner string, repo string) ([]*Git
 	return answer, nil
 }
 
+func extractRepositoryCommitAuthor(rc *github.RepositoryCommit) (gu *GitUser) {
+	commitAuthorPresent := rc.Commit.Author != nil
+	repositoryAuthorPresent := rc.Author != nil
+	repositoryAuthorOnly := !commitAuthorPresent && repositoryAuthorPresent
+	commitAndRepositoryAuthorMatch := commitAuthorPresent && repositoryAuthorPresent && rc.Commit.Author.GetLogin() == rc.Author.GetLogin()
+
+	if commitAuthorPresent || repositoryAuthorPresent {
+		gu = &GitUser{}
+	}
+
+	if commitAuthorPresent {
+		gu.Email = rc.Commit.Author.GetEmail()
+		gu.Login = rc.Commit.Author.GetLogin()
+		gu.Name = rc.Commit.Author.GetName()
+	}
+
+	if repositoryAuthorOnly {
+		gu.Email = rc.Author.GetEmail()
+		gu.Login = rc.Author.GetLogin()
+		gu.Name = rc.Author.GetName()
+		gu.URL = rc.Author.GetURL()
+		gu.AvatarURL = rc.Author.GetAvatarURL()
+	}
+
+	if commitAndRepositoryAuthorMatch {
+		gu.URL = rc.Author.GetURL()
+		gu.AvatarURL = rc.Author.GetAvatarURL()
+	}
+
+	return
+}
+
 func (p *GitHubProvider) asGitHubCommit(commit *github.RepositoryCommit) GitCommit {
 	message := ""
 	if commit.Commit != nil {
 		message = util.DereferenceString(commit.Commit.Message)
 	} else {
 		log.Logger().Warnf("No Commit object for for commit: %s", commit.GetSHA())
 	}
-	var author *GitUser
```

Comment:
👍 for extracting this into a function.

Reply:
BTW, there was another pull request for this just a short while back where I left this [comment](https://github.com/jenkins-x/jx/pull/5931#discussion_r338925448)